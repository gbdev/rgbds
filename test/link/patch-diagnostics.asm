def fzero equs "startof(\"test\")"
section "test", rom0
ld a, $8000_0000 / (fzero - 1)
ld a, $8000_0000 / (fzero - 2)
ld a, 1 << (fzero - 1)
ld a, 1 << (fzero + 32)
ld a, (fzero - 1) >> 1
ld a, 1 >> (fzero - 1)
ld a, 1 >> (fzero + 32)
ld a, 1 >>> (fzero - 1)
ld a, 1 >>> (fzero + 32)

; We rely on this landing at address $0000, which isn't *guaranteed*...
assert startof("test") == 0
