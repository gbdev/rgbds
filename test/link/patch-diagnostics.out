warning: patch-diagnostics.asm(10): [-Wshift-amount]
    Shifting right by large amount 32
warning: patch-diagnostics.asm(9): [-Wshift-amount]
    Shifting right by negative amount -1
warning: patch-diagnostics.asm(8): [-Wshift-amount]
    Shifting right by large amount 32
warning: patch-diagnostics.asm(7): [-Wshift-amount]
    Shifting right by negative amount -1
warning: patch-diagnostics.asm(6): [-Wshift]
    Shifting right negative value -1
warning: patch-diagnostics.asm(5): [-Wshift-amount]
    Shifting left by large amount 32
warning: patch-diagnostics.asm(4): [-Wshift-amount]
    Shifting left by negative amount -1
warning: patch-diagnostics.asm(3): [-Wdiv]
    Division of -2147483648 by -1 yields -2147483648
warning: patch-diagnostics.asm(3): [-Wtruncation]
    Value $80000000 (may be negative?) is not 8-bit
